cql/ManageRareAbnormality.cql

```diff
@@ -980,12 +1117,18 @@ define RecommendationForHistologyResults:
           'Treatment is recommended for histologic HSIL.'
         }
       }
+      
+// =====================================================      
     // I.2 Management of Histologic HSIL (CIN2 or CIN3)
     when ( // I2.1
       AgeInYears() >= 25 and
       not Dash.Pregnant and
-      MostRecentBiopsyReportWasWithinPastYear and
-      HistologyInterpretedAsCin3
+      AnyHistologyInterpretedAsCin3 and
+      (
+        Collate.DateOfLastCervicalPrecancerTreatment is null or 
+        not (Collate.DateOfLastCervicalPrecancerTreatment occurs after Collate.MostRecentCin3Biopsy.date) or
+        Collate.DateOfLastCervicalPrecancerTreatment 7 years or more before Collate.MostRecentCin3Biopsy.date
+      )
```

Comment:
Whether or not this logic is condensed according to the logic I suggested above, this logic should be abstracted out to it's own CQL expression. We should make some sort of "NotTreated" expression that includes the logic statements to make things more efficient, since it is repeated many times throughout section.

You could even get fancy and make it a function, that takes in a biopsy date as the parameter value, and returns true if there is no treatment 7 years before or anytime after that date. Doing something like this would make the actual recommendation logic more readable and would make it so that you don't have to change several different parts of the code if one piece of this logic needs to be tweaked.

Reply:
I really like the refactoring to make this a function! However, I believe the last line of the function needs a `not` before it, since you are checking whether a previous treatment did not exist or if the previous treatment **_was not_** in the time interval of 7 years before the lookback date to now.

CURRENT FUNCTION: 
```cql
define function HadNoCervicalPrecancerTreatmentFromDateToPresent(d System.DateTime, Lookback System.Quantity):
  Collate.DateOfLastCervicalPrecancerTreatment is null or 
  Collate.DateOfLastCervicalPrecancerTreatment in Interval[d - Lookback, Now()]
```

SUGGESTED CHANGE:
```cql
define function HadNoCervicalPrecancerTreatmentFromDateToPresent(d System.DateTime, Lookback System.Quantity):
  Collate.DateOfLastCervicalPrecancerTreatment is null or 
  not (Collate.DateOfLastCervicalPrecancerTreatment in Interval[d - Lookback, Now()])
```

Reply:
Thanks for catching this. This was an omission - I started with the function confirming the presence of treatment and then changed it to confirm absence of treatment. I left out the not( ) part.

Reply:
No problem. I did some local testing by tweaking an existing test file to check that the new functionality with the added `not` works, but it could be worth adding a test to address this case, since it didn't seem to get caught in any of the existing test cases. The CQL for this is resolved in https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/commit/798efb65d50eeacb95ff027494fe6bebc1d04660, so I'll resolve this comment thread.